ui/app/rif/actions/index.js

```diff
@@ -442,6 +443,18 @@ function navigateTo (screenName, params, resetNavigation = false) {
   return currentNavigation;
 }
 
+function resetNavigation () {
```

Comment:
Is there any way to dispatch this function only on the accountChange hook? That means, to invoke it only on one specific part of the code instead of on each situation?

Reply:
Unless you don't want to fix this problem for each case and just for the dropdown which i think doesn't make any sense then no it isn't, we have to reset navigation each time if we want to avoid the problem since nifty not only changes the account by the dropdown, some times you are importing accounts for example and that it's a rare case but it needs to be included.

Reply:
Yes exactly, i was thinking on account changes and import.

Reply:
I thought about this and i made a refactor that will only affect the nifty code in 2 points now, that's better than the other solution, basically if an account has changed it goes to home by default in nifty, but when you change network it doesn't. Now what i did is to go to home when the network has changed and it's RSK, then we reset the navigation when you enter the rif services, so we force the reset since the user is starting from home each time which it make sense also because if you are in RSK testnet and you change the network to RSK main you don't want to show the same page inside rif services because you don't know if the user can see it.